index.html

```diff
@@ -5156,17 +5198,91 @@ <h3>Overview</h3>
         and generating a JSON-LD document in expanded form for all
         <a>RDF literals</a>, <a>IRIs</a>
         and <a>blank node identifiers</a>.
-        If the <a data-link-for="JsonLdOptions">useNativeTypes</a> flag is set to <code>true</code>,
-        <a>RDF literals</a> with a
-        <a>datatype IRI</a>
-        that equals <code>xsd:integer</code> or <code>xsd:double</code> are converted
-        to a <a>JSON numbers</a> and <a>RDF literals</a>
-        with a <a>datatype IRI</a>
-        that equals <code>xsd:boolean</code> are converted to <code>true</code> or
-        <code>false</code> based on their
-        <a>lexical form</a>
-        as described in
-        <a class="sectionRef" href="#data-round-tripping"></a>.
+
+        If the <a data-link-for="JsonLdOptions">useNativeTypes</a> flag is set
+        to <code>true</code>, <a>RDF literals</a> are converted to JSON values
+        based on their <a>datatype IRI</a> and <a>lexical form</a>, as described
+        in <a class="sectionRef" href="#algorithm"></a>.
+      </p>
+
+      <table class="simple">
+        <thead>
+          <tr>
+            <th>RDF datatype</th>
+            <th>RDF Literal</th>
+            <th>JSON</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th>(<em>None</em>, implying <code>xsd:string</code>)</th>
+            <td rowspan="2">An RDF string literal</td>
+            <td rowspan="2">JSON string literal</td>
+          </tr>
+          <tr>
+            <th><code>xsd:string</code></th>
+          </tr>
+
+          <tr>
+            <th rowspan="3"><code>xsd:integer</code></th>
+            <td>Whole number in the <code>[-2<sup>53</sup> + 1, 2<sup>53</sup> + 1]</code> range, representable as per [[IEEE-754-2008]]</td>
+            <td><a>JSON Number</a></td>
+          </tr>
+
+          <tr>
+            <td>Whole number out of that range</td>
+            <td>JSON string literal</td>
+          </tr>
+
+          <tr>
+            <td>Any other value</td>
+            <td class="error">Throw <a data-link-for="JsonLdErrorCode">invalid JSON literal</a></td>
+          </tr>
+
+          <tr>
+            <th rowspan="4"><code>xsd:double</code></th>
+            <td>Real number in the <code>[-2<sup>53</sup> + 1, 2<sup>53</sup> + 1]</code> range, representable as per [[IEEE-754-2008]]</td>
+            <td><a>JSON Number</a></td>
+          </tr>
+
+          <tr>
+            <td>Real number out of that range</td>
+            <td>JSON string literal</td>
+          </tr>
+
+          <tr>
+            <td>
+              Non-numeric values:
+
+              <ul>
+                <li><code>"NaN"^^xsd:double</code>,</li>
+                <li><code>"+INF"^^xsd:double</code>,</li>
+                <li><code>"-INF"^^xsd:double</code></li>
+              </ul>
+            </td>
+            <td class="error" rowspan="2">Throw <a data-link-for="JsonLdErrorCode">invalid JSON literal</a></td>
+          </tr>
```

Comment:
Same as above; I don't think that is a good idea, see my comments in the algorithm.